### PR TITLE
Fix API 422 response for free plan

### PR DESCRIPTION
**Pull request by t3d3r3**
## Description

Discussed on slack with Austin.
Nomics requires per-page param for free tier. 

## Changes

- Added required per-page param to nomics crypto endpoint. Others are not available on the free tier.

## Steps to Test

1. Nomics free plan API key
2. Call the crypto endpoint
3. Will receiver 422 Response: Your plan requires that you provide a per-page argument. See http://docs.nomics.com/ for more information.

## Quality Assurance

- [ ] Ran `yarn changeset` if adapter source code was changed
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [ ] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [ ] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [ ] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

**Comment by austinborn** [comment]
Hey @t3d3r3 , thank you for contributing this! One suggestion is to turn this into an input parameter with `100` as a default, like so:
```
export const inputParameters: InputParameters = {
  ...,
  pageSize: {
    aliases: ['page', 'pageLength', 'perPage'],
    required: false,
    description: 'The maximum number of items to return per paginated response.',
    type: 'number',
    default: 100
  },
}
```
And then use it in the request body like this:
```
const params = {
    ids,
    convert,
    key: config.apiKey,
    'per-page': validator.validated.data.pageSize,
  }
```
How does that sound?

**Comment by t3d3r3** [comment]
Good point @austinborn, fixed.  

Please note however that in the meantime nomics removed the requirement for this field. I don't have any official info but the api works again without it.

File: .changeset/tricky-kangaroos-grow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@chainlink/nomics-adapter": patch
+---
+
+Fix API 422 response for free plan
```
Comment (austinborn) [inline .changeset/tricky-kangaroos-grow.md:5]:
```suggestion
Add per-page field to nomics crypto endpoint
```

**Comment by austinborn** [comment]
> Please note however that in the meantime nomics removed the requirement for this field. I don't have any official info but the api works again without it.

I think it's a positive change, and the footprint is relatively small (low risk). I say we merge it once we get the checks to run + pass.

**Comment by cds95** [comment]
@t3d3r3 any updates on this PR?

**Comment by t3d3r3** [comment]
@cds95 i tried but could not get the tests to run on my dev env. I just built and ran the container to confirm it's working. Maybe @austinborn can help?

**Comment by t3d3r3** [comment]
Fixed one issue, for the integration tests however afaik ticker calls require paid plan.

**Comment by cds95** [comment]
@t3d3r3 hey sorry for the delayed response.  So you should be able to just mock the API response to the ticker calls so that you don't have to pay.  You can look at examples from our integration tests that make use of the `nock` library.

**Comment by t3d3r3** [comment]
Hi @cds95, thing is i have a paid plan, this is not the issue. I am not able to get the integration tests to pass on my dev setup (other tests are ok). External requests are failing and i cannot figure out why. As i mentioned before, if i build the container it works fine... I will give it another try next week but please let me know if you have any suggestions.

**Comment by boxhock** [comment]
Hey @t3d3r3, do you have any updates on this PR? Were you able to get the tests running?

> Please note however that in the meantime nomics removed the requirement for this field. I don't have any official info but the api works again without it.

Does this mean that this change is not necessary anymore?

**Comment by t3d3r3** [comment]
> Does this mean that this change is not necessary anymore?

At least for now no, nomics reverted the requirement.

I tried rerunning the tests but they keep failing with this error:

> "message":"AdapterError: Nock: No match for request {\n  \"method\": \"GET\",\n  \"url\": \"https://api.nomics.com/v1/currencies/ticker?ids=BTC&convert=EUR&key=MY_KEY&per-page=100\",\n  \"headers\": {\n    \"accept\": \"application/json\",\n    \"cache-control\": \"no-cache, no-store, must-revalidate\",\n    \"pragma\": \"no-cache\",\n    \"content-type\": \"application/json\",\n    \"user-agent\": \"axios/0.24.0\"\n  }\n}","jobRunID":"1","params":{"endpoint":"crypto","resultPath":"price","base":"BTC","quote":"EUR","pageSize":100},"feedID":"BTC/EUR","url":"https:/api.nomics.com/v1/currencies/ticker"}

Same without the patch (master):

> "message":"AdapterError: Nock: No match for request {\n  \"method\": \"GET\",\n  \"url\": \"https://api.nomics.com/v1/currencies/ticker?ids=BTC&convert=EUR&key=MY_KEY\",\n  \"headers\": {\n    \"accept\": \"application/json\",\n    \"cache-control\": \"no-cache, no-store, must-revalidate\",\n    \"pragma\": \"no-cache\",\n    \"content-type\": \"application/json\",\n    \"user-agent\": \"axios/0.24.0\"\n  }\n}","jobRunID":"1","params":{"endpoint":"crypto","resultPath":"price","base":"BTC","quote":"EUR"},"feedID":"BTC/EUR","url":"https:/api.nomics.com/v1/currencies/ticker"}

I still cannot figure out why i cannot pass the tests locally. If anyone can help with this (at least run the nomics tests locally to confirm if the issue is on my end) we could push it as the page requirement might return, otherwise we can close it until that happens...

**Comment by boxhock** [comment]
@t3d3r3 It seems that the integration tests are not working for you on `master` because you are setting the API key. Without setting an API key, it will default to `"fake-api-key"`, which is the key used in the mocks.

Now, for the integration tests for this patch to work, you would have to modify the tests to add the `per-page` param in the mocks.

I hope that clarifies the integration test issues you've been seeing. I would like to request one more change to this PR:

```typescript
export const batchablePropertyPath = [{ name: 'base', limit: 100 }]
```

This should avoid us running into issues with batch warming as well.

**Comment by boxhock** [comment]
Hey @t3d3r3, it seems this PR has gone stale, so I am going to close this PR now.

If you would still like to contribute with this PR, please feel free to update and re-open this PR!

**Pull request closed without merging**